libraries/AP_Compass/AP_Compass.h

```diff
@@ -166,22 +177,22 @@ friend class AP_Compass_Backend;
     bool consistent() const;
 
     /// Return the health of a compass
-    bool healthy(uint8_t i) const { return _state[i].healthy; }
-    bool healthy(void) const { return healthy(get_primary()); }
+    bool healthy(uint8_t i) const { return _get_state(Priority(i)).healthy; }
+    bool healthy(void) const { return healthy(0); }
     uint8_t get_healthy_mask() const;
 
     /// Returns the current offset values
     ///
     /// @returns                    The current compass offsets in milligauss.
     ///
-    const Vector3f &get_offsets(uint8_t i) const { return _state[i].offset; }
-    const Vector3f &get_offsets(void) const { return get_offsets(get_primary()); }
+    const Vector3f &get_offsets(uint8_t i) const { return _get_state(Priority(i)).offset; }
```

Comment:
I think we should make it more clear in all the modified accessors what the argument is.  So leaving it as "i" makes it appear that it is still an instance number that is being provided but in fact it's a priority number.  Perhaps the argument should be renamed to "priority" or "pri" or something that makes it more clear that we are expecting a priority number.  Another way to do it would be to add a new typedef so the argument could then be something like, "get_offsets(priority_num p) const ... ".

Reply:
ah, nevermind, I guess this "i" really is an instance.

Reply:
Its not an instance, it is priority. As far as consumers of frontend are concerned all operations are via Priority, there is no concept of instance and priority like before. It is not possible for consumers to fetch details by instance number, that is reserved only for backends. "Priority()" call is a constructor which creates Priority type instance. This method was introduced, to make it compile time illegal to fetch anything by Instance or plain int by mistake. This ensures the coder understands that a priority number is expected.